rs/execution_environment/src/execution/call_or_task.rs

```diff
@@ -570,27 +569,23 @@ impl CallOrTaskHelper {
             }
             // Query methods only persist certain changes to the canister's state.
             CanisterCallOrTask::Query(_) => {
-                if output.wasm_result.is_ok() {
-                    match canister_state_changes
-                        .system_state_modifications
-                        .apply_changes(
-                            round.time,
-                            &mut self.canister.system_state,
-                            round.network_topology,
-                            round.hypervisor.subnet_id(),
-                            round.log,
-                        ) {
-                        Ok(_) => self.canister.system_state.canister_version += 1,
```

Comment:
If I understand correctly, we were incrementing the version here only for the ReplicatedQueries? (the rest of the questing is in the tread below)